src/guide/scaling-up/testing.md

```diff
@@ -1,32 +1,75 @@
 # Testing
 
-## Overview
+## Why Test?
+
+Automated tests help you and your team build complex Vue applications quickly and confidently by preventing regressions and encouraging you to break apart your application into testable functions, modules, classes, and components. As with any application, your new Vue app can break in many ways, and it's important that you can catch these issues and fix them before releasing.
 
-When it comes to building reliable applications, automated tests can play a critical role in an individual or team's ability to build new features, refactor code, and fix bugs with confidence.
+In this guide, we'll cover basic terminology and provide our recommendations on which tools to choose for your Vue 3 application.
 
-In a Vue application, there are different types of concerns that can be covered by automated tests:
+There is one Vue-specific section covering composables. See [Testing Composables](#testing-composables) below for more details.
 
-- **Logic**: verify non-UI rendering logic is implemented correctly. This typically involves application-wide business logic (e.g. state management stores) and utility functions.
+:::tip Start testing early!
+We recommend you begin writing tests as soon as you can. The longer you wait to add tests to your application, the more dependencies your application will have, and the harder it will be to start.
+:::
 
-- **Visual**: given certain input props / state, verify that an isolated component or component tree is rendering the correct visual output.
+## Testing Types
 
-- **Interaction**: simulate user behavior, such as a click or input, and verify that an isolated component or component tree renders correct, updated output.
+When designing your Vue application's testing strategy, you should leverage the following testing types:
 
-- **User Flow**: given a sequence of user interactions that are necessary to complete an actual user task, check whether the application as a whole is working as expected.
+- **Unit**: Checks that inputs to a given function, class, or composable are producing the expected output or side effects.
```

Comment:
**note**: I am not quite sure we should have a separation here between unit and component testing as component tests are considered a part of unit testing.

Reply:
I think there's two ways to test components:

1. By whitebox testing the component, where the `wrapper` is the subject under test. This is what I'd consider Unit Testing your components.
2. By blackbox testing the component, where the DOM is the subject under test. Testing Library and Cypress exclusively promote this blackbox-style testing (which we're referring to as "Component Testing")

Reply:
I still disagree with this definition. VTU allows you to perform proper blackbox testing if you don't assert `wrapper.vm`  properties. Hence, the line between component and unit testing is still blurred for me.

Reply:
I think you misunderstood me. You are correct.

Blackbox testing: DOM is the subject under test. VTU (but w/o accessing wrapper.vm or major stubbing), VTL (does not let you access internals), Cy (does not let you access internals)

Whitebox testing: Wrapper is the subject under test, DOM is _sometimes_ used. Lots of stubbing. VTU supports whitebox via Shallow Mount. Not possible to perform with Cy or VTL.